### ISSUE TITLE
Event handlers are not detached with changed selector in yii.gridView.js

### ISSUE THREAD

**Issue by arogachev** [yiisoft/yii2#13232]
Currently in `yii.gridView.js` we have 3 internal event handlers which are attached and detached in similar fashion:

```js
$(document).off('change.yiiGridView keydown.yiiGridView', settings.filterSelector)
    .on('change.yiiGridView keydown.yiiGridView', settings.filterSelector, function (event) {
    // Event handler's code
});
```

The problem of such approach lies in the selector which event handler is bind to.

Consider following initialization:

```js
var settings = {
    filterUrl: '/posts/index',
    filterSelector: '#w0-filters-input, #w0-filters-select'
};
$('#w0').yiiGridView(settings);
```

If we dynamically change selector and reinitialize the plugin with new settings:

```js
var newSettings = {
    filterUrl: '/posts/index',
    filterSelector: '#w0-filters-select'
};
$('#w0').yiiGridView(settings);
```

old event handlers will not be removed because `off` is always executed with current (new) selector:

```js
$(document).off('change.yiiGridView keydown.yiiGridView', settings.filterSelector)
```

`settings.filterSelector` will be `w0-filters-select` here.

As a result, interacting with input will cause the filter form to be submitted.

**Comment by samdark** [comment]
Is it fixed now?

**Comment by arogachev** [comment]
@samdark Yes, it's fixed in #13230.
